apps/server/src/modules/oauth/uc/oauth.uc.ts

```diff
@@ -1,136 +1,22 @@
-
-		const uuid: string = this.extractUUID(decodedToken);
-
-		const user: User = await this.findUserById(uuid, systemId);
-
-		const jwtResponse: string = await this.getJWTForUser(user);
-
-		const response: OAuthResponse = new OAuthResponse();
-		response.jwt = jwtResponse;
-		response.idToken = queryToken.id_token;
-		response.logoutEndpoint = system.oauthConfig.logoutEndpoint;
-
-		return response;
-	}
-
-	/**
-	 * @query query input that has either a code or an error
-	 * @return authorization code or throws an error
-	 */
-	checkAuthorizationCode(query: AuthorizationParams): string {
-		if (query.code) return query.code;
-		let errorCode = 'sso_auth_code_step';
-		if (query.error) {
-			errorCode = `sso_oauth_${query.error}`;
-			this.logger.error(`SSO Oauth authorization code request return with an error: ${query.code as string}`);
-		}
-		throw new OAuthSSOError('Authorization Query Object has no authorization code or error', errorCode);
-	}
-
-	async requestToken(code: string, system: System): Promise<OauthTokenResponse> {
-		const decryptedClientSecret: string = this.oAuthEncryptionService.decrypt(system.oauthConfig.clientSecret);
-		const tokenRequestPayload: TokenRequestPayload = TokenRequestPayloadMapper.mapToResponse(
-			system,
-			decryptedClientSecret,
-			code
-		);
-		const responseTokenObservable = this.httpService.post<OauthTokenResponse>(
-			tokenRequestPayload.tokenEndpoint,
-			QueryString.stringify(tokenRequestPayload.tokenRequestParams),
-			{
-				method: 'POST',
-				headers: {
-					'Content-Type': 'application/x-www-form-urlencoded',
-				},
-			}
-		);
-		const responseToken = await lastValueFrom(responseTokenObservable);
-		return responseToken.data;
-	}
-
-	async _getPublicKey(system: System): Promise<string> {
-		const client: JwksRsa.JwksClient = JwksRsa({
-			cache: true,
-			jwksUri: system.oauthConfig.jwksEndpoint,
-		});
-		const key: JwksRsa.SigningKey = await client.getSigningKey();
-		return key.getPublicKey();
-	}
-
-	async validateToken(idToken: string, system: System): Promise<IJWT> {
-		const publicKey = await this._getPublicKey(system);
-		const verifiedJWT = jwt.verify(idToken, publicKey, {
-			algorithms: ['RS256'],
-			issuer: system.oauthConfig.issuer,
-			audience: system.oauthConfig.clientId,
-		});
-		if (typeof verifiedJWT === 'string' || verifiedJWT instanceof String)
-			throw new OAuthSSOError('Failed to validate idToken', 'sso_token_verfication_error');
-		return verifiedJWT as IJWT;
-	}
-
-	extractUUID(decodedJwt: IJWT): string {
-		if (!decodedJwt || !decodedJwt.uuid) throw new OAuthSSOError('Failed to extract uuid', 'sso_jwt_problem');
-		const { uuid } = decodedJwt;
-		return uuid;
-	}
-
-	async findUserById(uuid: string, systemId: string): Promise<User> {
-		let user: User;
+		let oauthResponse: OAuthResponse;
 		try {
-			user = await this.userRepo.findByLdapId(uuid, systemId);
+			oauthResponse = await this.oauthService.processOauth(query, systemId);
 		} catch (error) {
```

Comment:
Why are you catching every error ? and why you dont handle it in service?

Reply:
we are catching errors in the service now, thanks 👍